src/Streamly/Internal/Data/Array/Prim/MutTypesInclude.hs

```diff
@@ -373,17 +370,25 @@ lpackArraysChunksOf n (Fold step1 initial1 extract1) =
     extract (Tuple' Nothing r1) = extract1 r1
     extract (Tuple' (Just buf) r1) = do
         r <- step1 r1 buf
-        extract1 r
+        case r of
+            FL.Partial rr -> extract1 rr
+            FL.Done _ -> return ()
+            FL.Done1 _ -> return ()
 
     step (Tuple' Nothing r1) arr = do
             len <- byteLength arr
             if len >= n
             then do
                 r <- step1 r1 arr
-                extract1 r
-                r1' <- initial1
-                return (Tuple' Nothing r1')
-            else return (Tuple' (Just arr) r1)
+                case r of
+                    FL.Done _ -> return $ FL.Done ()
+                    -- XXX Check this
+                    FL.Done1 _ -> return $ FL.Done1 ()
+                    FL.Partial s -> do
+                        extract1 s
+                        r1' <- initial1
```

Comment:
My guess is that since the fold has a `()` result we only care about the effects. And the current fold runs the `done` effect every time the accumulated fold's length crossed the limit.